src/snmp.c

```diff
@@ -1327,22 +1327,24 @@ static int csnmp_read_table(host_definition_t *host, data_definition_t *data) {
 
   status = 0;
   while (status == 0) {
-    int oid_list_todo_num;
-
     req = snmp_pdu_create(SNMP_MSG_GETNEXT);
     if (req == NULL) {
       ERROR("snmp plugin: snmp_pdu_create failed.");
       status = -1;
       break;
     }
 
-    oid_list_todo_num = 0;
+    size_t oid_list_todo_num = 0;
+    size_t var_idx[oid_list_len];
+    memset(&var_idx, 0, sizeof(var_idx));
```

Comment:
Sorry for yet another nit: When initializing arrays like that, we don't explicitly take the address of the array, i.e.: `memset(var_ids, 0, sizeof(var_idx));` without the `&`. (Both work, but we never use `&array` elsewhere …)

I really wish variable-length arrays could be zero-initialized with `{0}` …

Reply:
https://github.com/collectd/collectd/blob/master/src/tcpconns.c#L376
```
memset(&count_total, '\0', sizeof(count_total));
```
The declaration is:
```
static uint32_t count_total[TCP_STATE_MAX + 1];
```
:-)

Reply:
Updated, amended.